public/components/create_report/delivery/delivery.tsx

```diff
@@ -13,60 +13,205 @@
  * permissions and limitations under the License.
  */
 
-import React from 'react';
+import React, { useState } from 'react';
 import {
-    EuiFieldText,
-    EuiFormRow,
-    EuiPageHeader,
-    EuiTitle,
-    EuiPageContent,
-    EuiPageContentBody,
-    EuiHorizontalRule,
-    EuiTextArea,
-  } from '@elastic/eui';
+  EuiFieldText,
+  EuiFormRow,
+  EuiPageHeader,
+  EuiTitle,
+  EuiPageContent,
+  EuiPageContentBody,
+  EuiHorizontalRule,
+  EuiTextArea,
+  EuiCheckbox,
+  EuiSpacer,
+  EuiComboBox,
+  EuiText,
+  EuiLink,
+  EuiPopover,
+  EuiListGroup,
+} from '@elastic/eui';
 import { DeliveryRecipientsBox } from './delivery_recipients_box';
-import { DeliveryChannelRadio } from './delivery_channel_radio';
+import { EMAIL_RECIPIENT_OPTIONS } from './delivery_constants';
 
-const ReportDelivery = (props) => {
-  const { 
-    deliveryEmailSubject, 
-    onChangeDeliveryEmailSubject,
-    deliveryEmailBody,
-    onChangeDeliveryEmailBody
-  } = props;
+const INSERT_PLACEHOLDER_OPTIONS = [
+  {
+    label: 'Report details URL',
+    href: '#',
+    iconType: 'link',
+    size: 's',
+  },
+  {
+    label: 'Report source URL',
+    href: '#',
+    iconType: 'link',
+    size: 's',
+  },
+  {
+    label: 'File download URL',
+    href: '#',
+    iconType: 'link',
+    size: 's',
+  },
+  {
+    label: 'Report creation timestamp',
+    href: '#',
+    iconType: 'clock',
+    size: 's',
+  },
+];
 
-  return (
-    <EuiPageContent panelPaddingSize={"l"}>
-      <EuiPageHeader>
-        <EuiTitle>
-          <h2>Delivery</h2>
-        </EuiTitle>
-      </EuiPageHeader>
-      <EuiHorizontalRule/>
-      <EuiPageContentBody>
-        <EuiFormRow label="Channel" helpText="Define delivery notification channel">
-          <DeliveryChannelRadio/>
-        </EuiFormRow>
-        <EuiFormRow label="Recipients">
-          <DeliveryRecipientsBox/>
+export function ReportDelivery() {
+  const [emailCheckbox, setEmailCheckbox] = useState(false);
+  const [emailRecipients, setEmailRecipients] = useState([]);
+  const [emailSubject, setEmailSubject] = useState('');
+  const [emailBody, setEmailBody] = useState('');
+  const [includeReportAsAttachment, setIncludeReportAsAttachment] = useState(
+    false
+  );
+  const [insertPlaceholder, setInsertPlaceholder] = useState(false);
+
+  const onChangeEmailCheckbox = (e: {
+    target: { checked: React.SetStateAction<boolean> };
+  }) => {
+    setEmailCheckbox(e.target.checked);
+  };
+
+  const onChangeEmailRecipients = (e) => {
```

Comment:
Instead of `on` Prefix implementation with `handle` to differentiate the passing props and easy to find the implementation.